src/protect/addons/assess_immunotherapy_resistance.py

```diff
@@ -0,0 +1,127 @@
+    with open(input_files['immune_resistance_pathways.json']) as json_file:
+        json_data = json.load(json_file)
+
+    # Read patient file
+    patient_df = pd.read_csv('rsem_quant.tsv', sep=' ', delimiter='\t', header='infer', index_col=0)
+    patient_df.index = (patient_df.index).str.replace('\\..*$', '')
+
+    with open('immunotherapy_resistance_report.txt', 'w') as report_file:
+        # See if data exsits for specified tumor type
+        try:
+            pathways = json_data['Cancer_to_pathway'][tumor_type]
+        except KeyError:
+            report_file.write("Data not available")
+        else:
+            # If data exists, write a report
+            for pathway in pathways:
+                up_is_good = json_data['Pathways'][pathway]['up_is_good']
+
+                if up_is_good:
+                    comp_fn = lambda x, y: x >= y
+                else:
+                    comp_fn = lambda x, y: x < y
+
+                # Describe pathway and genes for it
+                report_file.write('Pathway: ' + pathway + '\n')
+                description = json_data['Pathways'][pathway]['description']
+                report_file.write('Description of pathway:\n ' + textwrap.fill(description, width=100)
+                                  + '\n')
+                report_file.write('Pathway genes: \n')
+                report_file.write('\t{:10}{:<20}{:<20}{:<12}\n'.format('Gene', 'Threshold_GTEX',
+                                                                       'Threshold_TCGA_N', 'Observed'))
+                status = []
+                # Write TCGA, GTEX, and observed values
+                for gene in json_data['Pathways'][pathway]['genes']:
+                    gtex = '{0:.2f}'.format(
+                            float(full_data.loc[gene, TCGAToGTEx[tumor_type]])) \
+                            if gene in full_data.index else 'NA'
+                    tcga = '{0:.2f}'.format(
+                            float(full_data.loc[gene, tumor_type + ' normal'])) \
+                            if gene in full_data.index else 'NA'
+
+                    tpmValue = '{0:.2f}'.format(float(patient_df.loc[gene, 'TPM'])) \
+                                if gene in patient_df.index else 'NA'
+                    ensg = json_data['Pathways'][pathway]['genes'][gene]
+                    report_file.write('\t{:10}{:<20}{:<20}{:<12}\n'.format(ensg, gtex, tcga, tpmValue))
+                    if gtex != 'NA' and tpmValue != 'NA':
+                        tcga_bool = comp_fn(float(tpmValue), float(tcga))
+                        gtex_bool = comp_fn(float(tpmValue), float(gtex))
+                        status.append(tcga_bool and gtex_bool)
+
+                # Based on the number of genes with expression values above normal, assess the status
+                if sum(status) >= 0.75 * len(status):
+                    report_file.write('\nStatus: Based on the results the patient is recommended for\
```

Comment:
We can't "recommend" anything. That would be practicing medicine without a license. Reword it so say "Patient might react well to the treatment given the tumor gene signatures"